specifications/Paxos/VotingApalache.tla

```diff
@@ -0,0 +1,125 @@
+------------------------------- MODULE VotingApalache -------------------------------
+
+(***********************************************************************************)
+(* This is a version of Voting.tla that can be analyzed by the `^Apalache^'        *)
+(* model-checker. Here are the differences compared to Voting.tla:                 *)
+(*                                                                                 *)
+(*     * We give concrete definitions for the constants                            *)
+(*                                                                                 *)
+(*     * We fix the number of ballots                                              *)
+(*                                                                                 *)
+(*     * We add the necessary type annotations on variables                        *)
+(*                                                                                 *)
+(*     * We rewrite SafeAt and ShowsSafeAt to avoid ranges of integers with        *)
+(*       non-constant bounds (which `^Apalache^' does not support).                *)
+(*                                                                                 *)
+(* We also give an inductive invariant that proves the consistency property. On a  *)
+(* desktop computer from 2022, `^Apalache^' takes 1 minute and 45 seconds to check *)
```

Comment:
Out of curiosity... Do you have numbers for larger models such as four processes?

Reply:
With 4 processes, 4 values, and 4 ballots it took 25 minutes. With just 2 values, but still 4 processes and 4 ballots, it took only 1 minute 30 seconds.